### PR TITLE
BCDA-1242: Update Auth Function for Key Pairs

### PR DESCRIPTION
**Pull request by embh**
### Fixes [BCDA-1242](https://jira.cms.gov/browse/BCDA-1242)
BCDA needs a function in the auth package for generating a key pair for encryption, and assigning that key pair to an ACO.

### Proposed changes:
Add function to generate a key pair and assign it to a system, which can be linked to an ACO by client ID.

### Change Details
* Adds function `GenerateSystemKeyPair` that creates a 2048 bit RSA key pair, saves the public key in the database, and returns the private key.

### Security Implications
The newly added `GenerateSystemKeyPair` function uses Go's `rsa` library to create the key pair, as used elsewhere in the BCDA application. The new function is not currently called anywhere except for its tests.

### Acceptance Validation
Added two tests:
* `TestGenerateSystemKeyPair` - confirms that key pair is created successfully
* `TestGenerateSystemKeyPair_AlreadyExists` - confirms that function will return an error if encryption keys already exist for the system

### Feedback Requested
I know we wanted to put this in the `auth` package, but it was not clear where. Open to suggestions on where the function should live. And any other feedback.

### CODE REVIEW

File: bcda/auth/models_test.go [hunk 1]
```diff
@@ -63,6 +66,81 @@ func (s *ModelsTestSuite) TestTokenCreation() {
...
+		s.FailNow(err.Error())
+	}
+	assert.Equal(s.T(), &privateKey.PublicKey, publicKey)
+
+	s.db.Unscoped().Delete(&pubEncrKey)
+	s.db.Unscoped().Delete(&system)
+	s.db.Unscoped().Delete(&group)
+}
+
+func (s *ModelsTestSuite) TestGenerateSystemKeyPair_AlreadyExists() {
```
Comment (dhgreene) [inline bcda/auth/models_test.go]: This is a great test of the expected database behavior.  Can/should we test that a valid key pair is generated?

File: bcda/auth/models.go [hunk 2]
```diff
@@ -63,3 +68,56 @@ func GetACOByClientID(clientID string) (models.ACO, error) {
 	}
 	return aco, err
 }
+
+/*
+ GenerateSystemKeyPair creates a keypair for a system, based on the provided client ID.
+ The public key is saved to the database and the private key is returned.
+*/
+func GenerateSystemKeyPair(clientID string) (string, error) {
```
Comment (dhgreene) [inline bcda/auth/models.go]: To use the key, we seem to be having to convert the `string` to a `[]byte`.  Would that be a better return type?

Comment (embh) [reply bcda/auth/models.go]: I was thinking about that. The CLI command function could convert the `[]byte` to `string` to print it out. 

Before I make the change, any preferences about the return type, @DeirdreHolub or @alexhova ?

Comment (alexhova) [reply bcda/auth/models.go]: no preference here!

File: bcda/auth/models.go [hunk 3]
```diff
@@ -63,3 +68,55 @@ func GetACOByClientID(clientID string) (models.ACO, error) {
...
+
+/*
+ GenerateSystemKeyPair creates a keypair for a system. The public key is saved to the database and the private key is returned.
+*/
+func GenerateSystemKeyPair(systemID uint) (string, error) {
+	db := database.GetGORMDbConnection()
+	defer database.Close(db)
+
+	var system models.System
+	err := db.Preload("EncryptionKeys").First(&system, systemID).Error
```
Comment (alexhova) [inline bcda/auth/models.go]: i think `Find()` makes more sense semantically here but i know we've talked this to death already

**Pull request merged**